sql/sql_expression_operations.cc

```diff
@@ -2505,14 +2505,22 @@ BoundSqlExpression
 CaseExpression::
 bind(SqlBindingScope & scope) const
 {
+    std::shared_ptr<ExpressionValueInfo> info;
+
     BoundSqlExpression boundElse;
-    if (elseExpr)
+    if (elseExpr) {
         boundElse = elseExpr->bind(scope);
+        info = boundElse.info;
+    }
 
     std::vector<std::pair<BoundSqlExpression, BoundSqlExpression> > boundWhen;
 
     for (auto & w: when) {
         boundWhen.emplace_back(w.first->bind(scope), w.second->bind(scope));
+        if (info)
+            info = std::make_shared<ORExpressionValueInfo>(info, boundWhen.back().second.info);
```

Comment:
If left and right are the same type, or if one is a strict subset of the other, we should combine them: for example, `CASE WHEN ... THEN 1 ELSE 0` should return an IntValueInfo, and `CASE WHEN ... THEN 0.5 ELSE 1` should return a float value info.